### DISCUSSION TITLE
Using nui to override functions in the vim.ui module

### DISCUSSION THREAD

**Discussion by VonHeikemen** [MunifTanjim/nui.nvim#69] [Ideas]
On neovim 0.6 there is this new module `vim.ui` (see [:help lua-ui](https://neovim.io/doc/user/lua.html#lua-ui)) which, if I understand correctly, allows plugin authors to rely on certain "components" to ask users for information. At the same time, the user can overwrite the functions on this module so the UI that is rendered can be whatever they want.

So for example, a plugin author can have code like this on their plugins.

```lua
vim.ui.input({prompt = 'Input:  '}, function(value) 
  -- some code
end)
```

And the user on their config can override `vim.ui.input`.

```lua
vim.ui.input = function(opts, on_confirm) 
  -- anything...
end
```

I think it would be a good idea to have documented somewhere how people can use `nui` to override the functions on `vim.ui`. Or even provide a module that does it.

**Comment by MunifTanjim** [comment]
Related discussion: https://github.com/MunifTanjim/nui.nvim/discussions/50

Related Plugin: https://github.com/stevearc/dressing.nvim

> I think it would be a good idea to have documented somewhere how people can use nui to override the functions 

I just added a Wiki page for `vim.ui`: 

https://github.com/MunifTanjim/nui.nvim/wiki/vim.ui

> Or even provide a module that does it.

I don't really wanna do that. Reasons described here: https://github.com/MunifTanjim/nui.nvim/discussions/50#discussioncomment-1737353

---

I hope this helps 😃

**Reply by VonHeikemen** [reply]
Thank you, this is good. It's exactly what I wanted. I was too lazy to write the input implementation myself.
